neuraxle/metaopt/random.py

```diff
@@ -114,6 +117,103 @@ def _split(self, data_inputs):
         return splitted_data_inputs
 
 
+class WalkForwardTimeSeriesCrossValidation(BaseCrossValidation):
+    """
+    This will change perform a walk forward cross validation by performing a forward rolling split.
+    All the training split will start at the start of the time series but the end will increase toward
+    the end at each forward step. For the validation split it will start after each training split and
+    can have a delay to test prediction according to some delay.
+
+    :param scoring_function: scoring function use to validate performance if it is not None, by default r2_score,
+    :param window_size: the size of the time step taken between each forward roll, by default 3.
+    :param initial_window_number: the size of the first training split length in term of number of window a window is defined by the window_size, by default 3.
+    :param window_delay_size: the size of the delay between the end of the training split and the validation split, by default 0.
+    :param joiner the joiner callable that can join the different result together.
+    :return: WalkForwardTimeSeriesCrossValidation instance.
+    """
+
+    def __init__(self, scoring_function=r2_score, window_size=3, initial_window_number=3, window_delay_size=0, joiner=NumpyConcatenateOnTimeStep()):
+        self.window_size = window_size
+        self.window_delay_size = window_delay_size
+        self.initial_window_number = initial_window_number
```

Comment:
I'm not sure to understand `window_delay_size` and `initial_window_number`. Is `window_delay_size` a padding between the train and val split? And is `initial_window_number` a multiple of `window_size` ?

For example, in the following image, would `initial_window_number = 5`, `window_size=1`, and `self.window_delay_size=0`, exactly ? 

![](https://robjhyndman.com/files/cv1-1.png)

Reply:
Note: After reading the rest of the code and commenting, it seems to me that my understanding is right, but please fix me if I'm wrong, otherwise resolve this :)

Reply:
window_delay_size is the padding between train and validation as in this image you are right.
![image](https://user-images.githubusercontent.com/50344651/62424291-f0a3db80-b69a-11e9-8577-e0a2f7f70aa8.png)

initial_window_number is an `int` (could be a float if wanted a not round number) that represent how many `windows` of `window_size` the first training block should have. I could change it to `initial_training_size` (and have it set in term of size and not in term of windows), `validation_delay_size` and `validation_size` if this is clearer.


For your exemple it will be : `initial_window_number = 6`, `window_size=1`, and `window_delay_size=0`.